pkg/k8shandler/fluentd.go

```diff
@@ -387,9 +411,9 @@ func (clusterRequest *ClusterLoggingRequest) updateFluentdDaemonsetIfRequired(de
 	desired, different := isDaemonsetDifferent(current, desired)
 
 	if different {
-
+		current.Spec = desired.Spec
```

Comment:
i think we lose generated k8s metadata by doing this (since `current` is populated from a `Get`)